plugins/scaffolder-backend/src/service/router.test.ts

```diff
@@ -515,186 +656,634 @@ describe('createRouter', () => {
+          .send({
+            templateRef: stringifyEntityRef({
+              kind: 'template',
+              name: 'create-react-app-template',
+            }),
+            values: {
+              required: 'required-value',
+            },
+          });
+
+        expect(response.body.id).toBe('a-random-id');
+        expect(response.status).toEqual(201);
+      });
+
+      it('should call the broker with a correct spec', async () => {
+        const broker =
+          taskBroker.dispatch as jest.Mocked<TaskBroker>['dispatch'];
+
+        await request(app)
+          .post('/v2/tasks')
+          .send({
+            templateRef: stringifyEntityRef({
+              kind: 'template',
+              name: 'create-react-app-template',
+            }),
+            values: {
+              required: 'required-value',
+            },
+          });
+        expect(broker).toHaveBeenCalledWith(
+          expect.objectContaining({
+            createdBy: 'user:default/guest',
+            secrets: {
+              backstageToken: 'token',
+            },
+
+            spec: {
+              apiVersion: mockTemplate.apiVersion,
+              steps: mockTemplate.spec.steps.map((step, index) => ({
+                ...step,
+                id: step.id ?? `step-${index + 1}`,
+                name: step.name ?? step.action,
+              })),
+              output: mockTemplate.spec.output ?? {},
+              parameters: {
+                required: 'required-value',
+              },
+              user: {
+                entity: mockUser,
+                ref: 'user:default/guest',
+              },
+              templateInfo: {
+                entityRef: stringifyEntityRef({
+                  kind: 'Template',
+                  namespace: 'Default',
+                  name: mockTemplate.metadata?.name,
+                }),
+                baseUrl: 'https://dev.azure.com',
+                entity: {
+                  metadata: mockTemplate.metadata,
                 },
-              ],
+              },
+            },
+          }),
+        );
+      });
+
+      describe('when the identity api throws an error', () => {
+        beforeEach(() => {
+          getIdentity.mockImplementation(
+            async ({
+              request: _request,
+            }: IdentityApiGetIdentityRequest): Promise<
+              BackstageIdentityResponse | undefined
+            > => {
+              throw new Error('whoops!');
+            },
+          );
+        });
+        it('return an error', async () => {
+          const response = await request(app)
+            .post('/v2/tasks')
+            .send({
+              templateRef: stringifyEntityRef({
+                kind: 'template',
+                name: 'create-react-app-template',
+              }),
+              values: {
+                required: 'required-value',
+              },
             });
-          });
+          expect(response.status).not.toEqual(201);
```

Comment:
This is unhandled and surfaces a 500 error to the user. I think we have three options:

- expect the request handler implementation to handle the error and return a 401 or 403.
- expect the backstage instance be configured with a middleware that validates the provided credentials and returns a 401 or 403.
- or do neither and surface a 500.

Reply:
awesome thank you 🙇